src/lavinmq/exchange/headers.cr

```diff
@@ -63,27 +71,19 @@ module LavinMQ
       end
     end
 
-    # ameba:disable Metrics/CyclomaticComplexity
-    private def matches(bindings, routing_key, headers, & : Queue | Exchange ->)
-      bindings.each do |bt, dst|
-        args = bt[1] || next
+    private def matches(headers) : Iterator(Destination) # , & : Destination ->)
+      @bindings.each.select do |args, _|
```

Comment:
```suggestion
      @bindings.each_key.select do |args|
```

Reply:
`each_key` gives a iterator of just they keys right? Because I need the values from the hash after the `.select`